source/Applications/Advanced/TransformViaArucoMarker/TransformViaArucoMarker.cpp

```diff
@@ -0,0 +1,308 @@
+            float b = (x - x1) / (x2 - x1); // NOLINT(cppcoreguidelines-narrowing-conversions)
+
+            const cv::Point3f fxy1{ a * q11.x + b * q21.x, a * q11.y + b * q21.y, a * q11.z + b * q21.z };
+            const cv::Point3f fxy2{ a * q12.x + b * q22.x, a * q12.y + b * q22.y, a * q12.z + b * q22.z };
+
+            float c = (y2 - y) / (y2 - y1); // NOLINT(cppcoreguidelines-narrowing-conversions)
+            float d = (y - y1) / (y2 - y1); // NOLINT(cppcoreguidelines-narrowing-conversions)
+            markerPoints3D.emplace_back(
+                cv::Point3f{ c * fxy1.x + d * fxy2.x, c * fxy1.y + d * fxy2.y, c * fxy1.z + d * fxy2.z });
+        }
+
+        return markerPoints3D;
+    }
+
+    Zivid::Matrix4x4 transformationMatrix(const cv::Matx33d &rotationMatrix, const cv::Vec3d &translationVector)
+    {
+        auto transformMatrix = Zivid::Matrix4x4{ 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1 };
+
+        for(size_t row = 0; row < rotationMatrix.rows; row++)
+        {
+            for(size_t col = 0; col < rotationMatrix.cols; col++)
+            {
+                transformMatrix(row, col) = rotationMatrix(row, col);
+            }
+        }
+
+        transformMatrix(0, 3) = translationVector[0];
+        transformMatrix(1, 3) = translationVector[1];
+        transformMatrix(2, 3) = translationVector[2];
+
+        return transformMatrix;
+    }
+
+    void displayMarkerPose(const Zivid::PointCloud &pointCloud,
+                           const std::vector<cv::Point2f> &corners2D,
+                           const std::vector<cv::Point2f> &center2D)
+    {
+        auto rgbaImage = cv::Mat(pointCloud.height(), pointCloud.width(), CV_8UC4); // NOLINT(hicpp-signed-bitwise)
+        auto *ptr =
+            reinterpret_cast<Zivid::ColorRGBA *>(rgbaImage.data); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
+        pointCloud.copyData(ptr);
+        auto bgr = cv::Mat(pointCloud.height(), pointCloud.width(), CV_8UC4); // NOLINT(hicpp-signed-bitwise)
+        cv::cvtColor(rgbaImage, bgr, cv::COLOR_BGR2RGB);
+
+        auto poseCenter = center2D[0];
+        auto xAxisPoint = cv::Point2f(((corners2D[3].x + corners2D[0].x) / 2), ((corners2D[3].y + corners2D[0].y) / 2));
+        auto yAxisPoint = cv::Point2f(((corners2D[2].x + corners2D[3].x) / 2), ((corners2D[2].y + corners2D[3].y) / 2));
+
+        cv::circle(bgr, poseCenter, 2, cv::Scalar(255, 0, 0), 2);
+        cv::arrowedLine(bgr, poseCenter, xAxisPoint, cv::Scalar(0, 0, 255), 2);
+        cv::arrowedLine(bgr, poseCenter, yAxisPoint, cv::Scalar(0, 255, 0), 2);
+
+        cv::imshow("Aruco marker pose", bgr);
+        cv::waitKey(0);
+    }
+
+    Zivid::Matrix4x4 getMarkerPose(const Zivid::PointCloud &pointCloud,
+                                   const aruconano::Marker &markers,
+                                   const bool display = false)
+    {
+        const auto corners2D = get2DMarkerCorners(markers);
```

Comment:
Add a comment sentence about what is being done in this function and how, if possible?
Or add a number of smaller comments from line 190 to line 212 for each operation?

Reply:
done